src/main/java/org/jenkinsci/plugins/casc/Configurator.java

```diff
@@ -45,11 +45,11 @@
  *
  * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
  */
-public abstract class Configurator<T> implements ExtensionPoint {
+public abstract class Configurator<T> implements ExtensionPoint, ElementConfigurator<T> {
 
     private final static Logger logger = Logger.getLogger(Configurator.class.getName());
 
-
+    @CheckForNull
     public static RootElementConfigurator lookupRootElement(String name) {
```

Comment:
Probably should be `Optional<...>` as a return type, so we could use `orElse` to throw exceptions instead of verbosive `if configurator == null`

Reply:
Maybe, let's wait for what the maintainers say